config/src/config_wrapper.cpp

```diff
@@ -18,26 +18,198 @@
+        catch (const std::exception &e) {                                                          \
+            log << e.what();                                                                       \
+            return default;                                                                        \
+        }                                                                                          \
+    }
+
+#define LIST_GETTER(p_type, list_type, from, key)                                                  \
+    void key(int32_t num, p_type arr)                                                              \
+    {                                                                                              \
+        if (!loaded) {                                                                             \
+            load();                                                                                \
+        }                                                                                          \
+        try {                                                                                      \
+            auto tmp = from.get<std::vector<list_type>>(#key);                                     \
+            for (int i = 0; i < num; i++) {                                                        \
+                arr[i] = tmp.at(i);                                                                \
+            }                                                                                      \
+        }                                                                                          \
+        catch (const std::exception &e) {                                                          \
+            log << e.what();                                                                       \
+        }                                                                                          \
+    }
+
+bool loaded = false;
 
 config::Config static_config;
+config::Config dynamic_config;
+
+void load()
+{
+    if (loaded) {
+        return;
+    }
+
+    static_config.load_from_file("static_config.json");
+    dynamic_config.load_from_file("dynamic_config.json");
+
+    loaded = true;
+}
+
+CONFIG_GETTER(int32_t, int, static_config, number_of_stations, 0);
+CONFIG_GETTER(int32_t, int, static_config, number_of_end_stations, 0);
+CONFIG_GETTER(int32_t, int, static_config, number_of_robots, 0);
+CONFIG_GETTER(int32_t, int, static_config, number_of_waypoints, 0);
+CONFIG_GETTER(int32_t, int, static_config, waypoint_passing_time, 0);
+CONFIG_GETTER(int32_t, int, static_config, station_passing_time, 0);
+CONFIG_GETTER(double, double, static_config, uncertainty, 0.0);
+
+CONFIG_GETTER(int32_t, int, dynamic_config, current_station, 0);
+CONFIG_GETTER(int32_t, int, dynamic_config, current_waypoint, 0);
+CONFIG_GETTER(int32_t, int, dynamic_config, destination, 0);
```

Comment:
Hvorfor er det vi skal have en default værdi med som parameter? Kan den ikke pakkes ind i funktionen? Umildbart kan jeg kun se en forskel ved double?

Reply:
Hvis `CONFIG_GETTER` nu skal læse en streng? Men nej, der er i praksis ingen forskel, men CONFIG_GETTER er også blevet meget mindre på #60.

Reply:
Det lyder som coding in anticipation of need. Men lad os se på det til #60